etl/lib/mip.py

```diff
@@ -0,0 +1,203 @@
+"""
+Purpose: This script implements maximum intensity projections (MIP). This process involves
+taking 3D brain scans and compressing their maximum values down into a single 2D array.
+"""
+
+
+# TODO: preprocess coronal and sagittal scans so they have mips too
+
+import io
+import logging
+
+import numpy as np
+from google.cloud import storage
+from tensorflow.python.lib.io import file_io
+import imageio
+
+
+def authenticate():
+    return storage.Client.from_service_account_json(
+        './credentials/client_secret.json'
+    )
+
+
+def configure_logger():
+    root_logger = logging.getLogger()
+    root_logger.setLevel(logging.INFO)
+    handler = logging.StreamHandler()
+    formatter = logging.Formatter(
+        fmt='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    handler.setFormatter(formatter)
+    root_logger.addHandler(handler)
+
+
+def mip_array(array: np.ndarray, type: str) -> np.ndarray:
+    print(np.max(array, axis=0).shape)
+    return np.max(array, axis=0)
+
+
+def crop(arr: np.ndarray, whence: str):
```

Comment:
Im wondering if this file should be split. One for transformations (which would be better fit for transforms.py), and one for GCS management (download / uploading files). Maybe one more for __main__.